_blog/2018/v1alpha3-routing.md

```diff
@@ -0,0 +1,347 @@
+This results in a least common denominator that uses the most basic HTTP routing and discards every other feature of the
+proxy, which end up in annotations that are not portable.
+
+So the `Gateway` effort started. Istio `Gateway` overcomes the `Ingress` shortcomings by separating the L4-L6 spec from L7.
+It only configures the L4-L6 functions (e.g., ports to expose, TLS configuration) that are uniformly implemented by all good
+L7 proxies. Users can then use standard Istio route rules to control HTTP requests & TCP traffic entering a `Gateway` by
+binding a `VirtualService` to it. The route rules will be applied at the gateway if and only if one of the `VirtualService's`
+hosts corresponds to one of the hosts configured in the `Gateway`.
+
+For example, the following simple `Gateway` could be used to allow external https traffic to the bookinfo.com
+`VirtualService`, described earlier:
+
+```yaml
+apiVersion: networking.istio.io/v1alpha3
+kind: Gateway
+metadata:
+  name: bookinfo-gateway
+spec:
+  servers:
+  - port:
+      number: 443
+      name: https
+      protocol: HTTPS
+    hosts:
+    - bookinfo.com
+    tls:
+      mode: SIMPLE
+      serverCertificate: /tmp/tls.crt
+      privateKey: /tmp/tls.key
+```
+
+In this example, the `hosts` field specifies bookinfo.com concretely, but in general the binding can be much more flexible.
+A single `Gateway` can bind to multiple `VirtualServices` or a single `VirtualService` can be exposed on more than one
+`Gateway`.
+
+To configure the corresponding routes, the bookinfo `VirtualService` needs to be bound to the `Gateway` by adding an
+additional `gateways` field to the configuration:
+
+```yaml
+apiVersion: networking.istio.io/v1alpha3
+kind: VirtualService
+metadata:
+  name: bookinfo
+spec:
+  hosts:
+    - bookinfo.com
+  gateways:
+  - bookinfo-gateway # <---- bind to gateway
+  http:
+  - match:
+    - uri:
+        prefix: /reviews
+    route:
+    ...
+```
+
+Although primarily used as an ingress allowing external traffic into the mesh, a `Gateway` can also act as an egress to
+allow traffic in the mesh to exit. A `Gateway` can also model a proxy that is entirely internal to the mesh and implemented
+either by sidecars or by a middle proxy.  Ingress and egress gateways can also expose non HTTP services with the same
+ease.
+
+A `Gateway` simply configures a load balancer, regardless of where it will be running. Any number of gateways can exist
+within the mesh and multiple different gateway implementations can co-exist. In fact, a gateway configuration can be bound
+to a particular workload by specifying the set of workload (pod) labels as part of the configuration, allowing users to
+reuse off the shelf network appliances by writing a simple gateway controller. `Gateway` is much more general purpose than
+simply an ingress replacement.
+
+## Summary
+
+The Istio `v1alpha3` routing API is significantly more functional than its predecessor, but unfortunately not backwards
```

Comment:
tone nit: instead of emphasizing `functional`, emphasize `functionality`. v1alpha1 was functional, it just didn't have as many features.